plugins/importexport/csv/CSVImportExportPlugin.inc.php

```diff
@@ -261,11 +475,37 @@ function executeCLI($scriptName, &$args) {
 	 * Display the command-line usage information
 	 */
 	function usage($scriptName) {
-		echo __('plugins.importexport.csv.cliUsage', array(
+		echo __('plugins.importexport.csv.cliUsage', [
 			'scriptName' => $scriptName,
 			'pluginName' => $this->getName()
-		)) . "\n";
+		]) . "\n";
 	}
-}
-
 
+	private function requiredFieldsPresent($row)
+	{
+		[
+			$pressPath,
+			$authorString,
+			$title,
+			$abstract,
+			$seriesPath,
+			$year,
+			$isEditedVolume,
+			$locale,
+			$filename,
+			$doi,
+			$keywords,
+			$subjects,
+			$bookCoverImageName,
+			$bookCoverImageAltText,
+			$categories
+		 ] = array_pad($row, 15, null);
```

Comment:
The constant 15 spread through the code and the duplicated fields, makes me think that this should be changed.

In general I prefer such approach:
```php
$expectedHeaders = [
	'pressPath',
	'authorString',
	'title',
	'abstract',
	'seriesPath',
	'year',
	'isEditedVolume',
	'locale',
	'filename',
	'doi',
	'keywords',
	'subjects',
	'bookCoverImageName',
	'bookCoverImageAltText',
	'categories',
];
$header = $file->fgetcsv();

$missingHeaders = array_diff($expectedHeaders, $header);
if (count($missingHeaders)) {
	print_r($missingHeaders);
	exit;
}

foreach ($file as $fields) {
	$data = (object) array_combine($header, array_pad(array_map('trim', $fields), count($header), null));
    echo $data->pressPath;
}
```

Then you'll have less variables (avoid name conflicts) and can pass the `$data` object around.

For the validations below, does the CSV return null or an empty string for the empty cells? I'd guess it's an empty string, if I'm not wrong, then the `is_null()` should be replaced by `strlen()`.